apps/token/src/app-loader.tsx

```diff
@@ -17,6 +17,18 @@ import {
 import { useContracts } from './contexts/contracts/contracts-context';
 import { useRefreshAssociatedBalances } from './hooks/use-refresh-associated-balances';
 import { Connectors } from './lib/vega-connectors';
+import { useSearchParams } from 'react-router-dom';
+
+const useVegaWalletEagerConnect = () => {
+  const vegaConnecting = useEagerConnect(Connectors);
+  const { pubKey, connect } = useVegaWallet();
+  const [searchParams] = useSearchParams();
+  const [query] = React.useState(searchParams.get('address'));
```

Comment:
**thought (blocking)**: I think we need to make this check that address is a valid public key. I chose to be difficult and use `network`, which is valid, but I think at least requiring it is x length and valid makes sense?